lib/core/src/Cardano/Pool/Metrics.hs

```diff
@@ -163,15 +163,19 @@ monitorStakePools
     -> NetworkLayer IO t Block
     -> DBLayer IO
     -> IO ()
-monitorStakePools tr nl DBLayer{..} = do
+monitorStakePools tr nl db@DBLayer{..} = do
     cursor <- initCursor
     logInfo tr $ mconcat
-        [ "Start monitoring stake pools. Currently at "
+        [ "Monitoring stake pools. Currently at "
         , case cursor of
             [] -> "genesis"
             _  -> pretty (last cursor)
         ]
-    follow nl tr cursor forward backward header
+    follow nl tr cursor forward header >>= \case
+        Nothing    -> pure ()
+        Just point -> do
+            liftIO . atomically $ rollbackTo point
+            monitorStakePools tr nl db
```

Comment:
Like this `atomically $ rollbackTo point` and `initPoolProductionCursor` of the next `monitorStakePools` call are in different DB transactions. 

But… actually, that should impossibly be a problem.